api/handlers_api.go

```diff
@@ -10,210 +10,205 @@ import (
+		resp.AddError(err.Error(), cerr.ConstErrReady)
+		return resp
 	}
 
-	player.SetReady(readyPlayerReq.Payload.DefenceGrid)
-	return resp, game
+	return resp
 }
 
 // Handle the attack logic for the incoming request
-func (r *Request) HandleAttack() (mc.Message[mc.RespAttack], *mb.Player) {
+func (r Request) HandleAttack(game *mb.Game, attacker mb.Player, defender mb.Player, gm mb.GameManager) mc.Message[mc.RespAttack] {
 	var reqAttack mc.Message[mc.ReqAttack]
 	resp := mc.NewMessage[mc.RespAttack](mc.CodeAttack)
 
-	if err := json.Unmarshal(r.Payload, &reqAttack); err != nil {
+	if err := json.Unmarshal(r.payload, &reqAttack); err != nil {
 		resp.AddError(err.Error(), cerr.ConstErrInvalidPayload)
-		return resp, nil
-	}
-
-	game, attacker, err := r.Session.GameManager.FindGameAndPlayer(reqAttack.Payload.GameUuid, reqAttack.Payload.PlayerUuid)
-	if err != nil {
-		resp.AddError(err.Error(), cerr.ConstErrAttack)
-		return resp, nil
+		return resp
 	}
 
 	coordinates := mb.NewCoordinates(reqAttack.Payload.X, reqAttack.Payload.Y)
-	if !game.AreIncomingCoordinatesValid(coordinates) {
+	if !gm.AreAttackCoordinatesValid(game, coordinates) {
 		resp.AddError(cerr.ErrXorYOutOfGridBound(coordinates.X, coordinates.Y).Error(), cerr.ConstErrAttack)
-		return resp, nil
+		return resp
 	}
 
 	// Attack validity check
-	if !attacker.IsTurn {
-		resp.AddError(cerr.ErrNotTurnForAttacker(attacker.Uuid).Error(), cerr.ConstErrAttack)
-		return resp, nil
+	if !attacker.IsTurn() {
+		resp.AddError(cerr.ErrNotTurnForAttacker(attacker.GetUuid()).Error(), cerr.ConstErrAttack)
+		return resp
 	}
-	if attacker.DidAttackThisCoordinatesBefore(coordinates) {
+
+	if !attacker.IsAttackGridEmptyInCoordinates(coordinates) {
 		resp.AddError(cerr.ErrAttackPositionAlreadyFilled(coordinates.X, coordinates.Y).Error(), cerr.ConstErrAttack)
-		return resp, nil
+		return resp
 	}
 
-	// Idenitify the defender
-	defender := game.GetOtherPlayer(attacker)
-
-	if defender.AreCoordinatesAlreadyHit(coordinates) {
+	if defender.IsDefenceGridAlreadyHitInCoordinates(coordinates) {
 		resp.AddError(cerr.ErrDefenceGridPositionAlreadyHit(coordinates.X, coordinates.Y).Error(), cerr.ConstErrAttack)
-		return resp, defender
+		return resp
 	}
 
-	attacker.IsTurn = false
-	defender.IsTurn = true
+	attacker.SetTurnFalse()
+	defender.SetTurnTrue()
+
+	if defender.IsAttackMiss(coordinates) {
+		attacker.SetAttackGridToMiss(coordinates)
 
-	if defender.IsIncomingAttackMiss(coordinates) {
-		attacker.AttackGrid[coordinates.X][coordinates.Y] = mb.PositionStateAttackGridMiss
 		resp.AddPayload(mc.RespAttack{
 			X:               coordinates.X,
 			Y:               coordinates.Y,
 			PositionState:   mb.PositionStateAttackGridMiss,
-			SunkenShipsHost: game.HostPlayer.SunkenShips,
-			SunkenShipsJoin: game.JoinPlayer.SunkenShips,
-			IsTurn:          attacker.IsTurn,
+			SunkenShipsHost: gm.GetHostPlayerSunkenShips(game),
+			SunkenShipsJoin: gm.GetJoinPlayerSunkenShips(game),
```

Comment:
Same comment here. This could be written as `game.hostPlayer.sunkenShips`

Reply:
same as uuid, This field is private, It needs a getter unless we expose it publicly or define a getter method